nimare/annotate/gclda.py

```diff
@@ -405,14 +635,85 @@ def _update(self, loglikely_freq=1):
             is 1 (log-likelihood is updated every iteration).
         """
         self.iter += 1  # Update total iteration count
-
         LGR.debug(f"Iter {self.iter:04d}: Sampling z")
         self.seed += 1
-        self._update_word_topic_assignments(self.seed)  # Update z-assignments
+        (
+            self.topics["wtoken_topic_idx"],
+            self.topics["n_word_tokens_word_by_topic"],
+            self.topics["total_n_word_tokens_by_topic"],
+            self.topics["n_word_tokens_doc_by_topic"],
+        ) = _update_word_topic_assignments(
+            self.topics["wtoken_topic_idx"],
+            self.topics["n_word_tokens_word_by_topic"],
+            self.topics["total_n_word_tokens_by_topic"],
+            self.topics["n_word_tokens_doc_by_topic"],
+            self.topics["n_peak_tokens_doc_by_topic"],
+            np.array(self.data["wtoken_doc_idx"]),
+            np.array(self.data["wtoken_word_idx"]),
+            len(self.vocabulary),
+            self.params["beta"],
+            self.params["gamma"],
+            self.seed,
+        )  # Update z-assignments
 
         LGR.debug(f"Iter {self.iter:04d}: Sampling y|r")
         self.seed += 1
-        self._update_peak_assignments(self.seed)  # Update y-assignments
+        old_regions, old_topics = self.topics["peak_region_idx"], self.topics["peak_topic_idx"]
+        peak_probs = self._get_peak_probs(self)  # Retrieve p(x|r,y) for all subregions
+        np.random.seed(self.seed)  # Seed random number generator
+
+        # When the size of X is large and the task is fast, Parallel()(delayed(f)(x) for x in X)
+        # is very slow. Split data into smaller chuncks to speed up Parallel().
+        n_ptoken = len(self.data["ptoken_doc_idx"])
+        n_chunks = self.n_cores
+        chunk_size = math.ceil(n_ptoken / n_chunks)
+        # Array of seeds for multi-core reproducibility
+        seeds = np.random.randint(1e8, size=n_ptoken)
+        chunks = [
+            [
+                self.data["ptoken_doc_idx"][chunk_i : chunk_i + chunk_size],
+                old_regions[chunk_i : chunk_i + chunk_size],
+                old_topics[chunk_i : chunk_i + chunk_size],
+                peak_probs[chunk_i : chunk_i + chunk_size, :, :],
+                seeds[chunk_i : chunk_i + chunk_size],
+            ]
```

Comment:
can this be a dictionary so that the arguments being sent to `_update_peak_assignments` are explicit?